var/spack/repos/builtin/packages/py-pycuda/package.py

```diff
@@ -14,13 +14,28 @@ class PyPycuda(PythonPackage):
     homepage = "https://mathema.tician.de/software/pycuda/"
     pypi = "pycuda/pycuda-2019.1.2.tar.gz"
 
+    version('2020.1', sha256='effa3b99b55af67f3afba9b0d1b64b4a0add4dd6a33bdd6786df1aa4cc8761a5')
     version('2019.1.2', sha256='ada56ce98a41f9f95fe18809f38afbae473a5c62d346cfa126a2d5477f24cc8a')
     version('2016.1.2', sha256='a7dbdac7e2f0c0d2ad98f5f281d5a9d29d6673b3c20210e261b96e9a2d0b6e37')
 
+    @run_before('build')
+    def configure(self):
+        pyver = self.spec['python'].version.up_to(2).joined
+        boostlib = 'boost_python{0}'.format(pyver)
+        configure_args = [
+            '--no-use-shipped-boost',
+            '--boost-inc-dir={0}'.format(self.spec['boost'].prefix.include),
+            '--boost-lib-dir={0}'.format(self.spec['boost'].prefix.lib),
+            '--boost-python-libname={0}'.format(boostlib)
+        ]
+        python('configure.py', *configure_args)
+
     depends_on('py-setuptools', type='build')
     depends_on('cuda')
-    depends_on('boost')
+    depends_on('boost+python')
+    depends_on('python@3.6:', type=('build', 'run'), when='@2020.1:')
     depends_on('py-numpy@1.6:', type=('build', 'run'))
+    depends_on('py-pytools@2011.2:', type=('build', 'run'))
     depends_on('py-six', type='run')
```

Comment:
Six is no longer needed, other deps should be type=build/run

Reply:
six is still needed at runtime:
```shell
[aweits@localhost pycuda]$ grep -r "import six" *
examples/download-examples-from-wiki.py:    import six.moves.xmlrpc_client
pycuda/curandom.py:import six
pycuda/driver.py:import six
pycuda/elementwise.py:import six
pycuda/gpuarray.py:import six
pycuda/gpuarray.py:import six.moves.copyreg
pycuda/scan.py:import six
pycuda/tools.py:import six

```